ws-client/src/transport.rs

```diff
@@ -323,7 +310,7 @@ impl From<soketto::connection::Error> for WsConnectError {
 }
 
 /// Helper to parse an URL to a WebSocket address.
-pub fn parse_url(url: impl AsRef<str>) -> Result<(Vec<SocketAddr>, Host, Mode), WsHandshakeError> {
+pub fn parse_url(url: impl AsRef<str>) -> Result<(Vec<SocketAddr>, Host, Mode, Path), WsHandshakeError> {
```

Comment:
4 is about the limit for tuples imo. It's fine here but if there are ever more items added here I think we might want a `PeerConfig` struct or something.